docs/token-guides/wsteth-bridging-guide.md

```diff
@@ -0,0 +1,353 @@
+
+#### R-6: Dedicated upgradable bridge instances
+
+Deploy dedicated upgradable instances of bridges on L1 and L2.
+This is required to lay the foundation for the emergency capabilities (R-7) and for possible bridging of rebasable stETH. For more details on why, see for section [Why this guide needed](#why-is-this-guide-needed). For the architecture outline, see section [Reference architecture and permissions setup](#reference-architecture-and-permissions-setup).
+
+#### R-7: Pausable deposits and withdrawals
+
+To provide the capability to react fast and reduce losses in case of a security contingency, depositing and withdrawing should be pausable. Namely:
+
+- L1 bridge has pausable and resumable deposits;
+- L2 bridge/token has pausable and resumable withdrawals.
+
+The capability to pause should be granted both to the DAO Agent instance (or its L2 representation) and the dedicated Emergency Multisig which is [ratified by the Lido DAO](https://snapshot.org/#/lido-snapshot.eth/proposal/0xfe2a6a6506a642b616118363bc29aa83dd9ef2ec80447bb607a8f52c0a96aed0). The permission to resume is proposed to be granted only to the Lido DAO Agent (or its L2 representation).
+
+To reduce the power of the multisig beyond necessary, it is proposed to use "Gate Seals" mechanic. The mechanic limits the pause duration and restricts the capability to pause to a single time. To grant the capability repeatedly, the Lido DAO vote is required. The mechanic has been implemented, e.g., for withdrawals in Lido protocol on Ethereum in two parts:
+- one-time disposable pauser contact [Gate Seals](https://github.com/lidofinance/gate-seals);
+- [PausableUntil](https://github.com/lidofinance/lido-dao/blob/master/contracts/0.8.9/utils/PausableUntil.sol) contract (inherited by [WithdrawalQueue](https://github.com/lidofinance/lido-dao/blob/master/contracts/0.8.9/WithdrawalQueue.sol)).
+
+#### R-8: Support of ERC-2612 permit enhanced with EIP-1271
+
+The bridged wstETH should support [EIP-2612 permit ERC-20 token extension](https://eips.ethereum.org/EIPS/eip-2612) with [EIP-1271 standard signature validation method for contracts](https://eip1271.io/). The latter paves the way to Account Abstraction adoption, see https://eip1271.io/.
+
+Please take into account that the [OpenZeppeling ERC20 with permit (EIP-2612) implementation](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC20Permit.sol) does not support smart contract signatures validation EIP-1271 and thus shouldn't be used as it is. Please consider extending ERC20Permit using [OpenZeppeling SignatureChecker util](https://docs.openzeppelin.com/contracts/4.x/api/utils#SignatureChecker) or [stETHPermit contract](https://github.com/lidofinance/lido-dao/blob/master/contracts/0.4.24/StETHPermit.sol) as a reference implementation. NB, that the wstETH token itself on Ethereum doesn't support this due to non-upgradability.
```

Comment:
mention someway that it might be enabler for 4337-compatible Account Abstraction and Paymasters